ledger/src/blockstore_db.rs

```diff
@@ -773,8 +774,14 @@ impl Rocks {
 
 pub trait Column {
     type Index;
+    const KEY_LEN: usize;
```

Comment:
This is a tricky, but `KEY_LEN` may not always equal `mem::size_of::<Self::Index>()`. `mem::size_of()` will include any alignment padding bytes. On the other hand, we pack the key byte arrays end-to-end (no padding).

One example where the default would be wrong is here:
https://github.com/anza-xyz/agave/blob/d3fe8f8a01725e02c9ca594ffe8b1461f10b0e9a/ledger/src/blockstore_db.rs#L1299-L1300

Because of `Slot` (which is a u64), the struct must be aligned on an 8-byte stride so `mem::size_of::<MerkleRootMeta>() == 16`. But, our keys are `8 + 4 = 12` bytes (size of u64 + size of u32):
https://github.com/anza-xyz/agave/blob/d3fe8f8a01725e02c9ca594ffe8b1461f10b0e9a/ledger/src/blockstore_db.rs#L1309-L1310